contracts/ExampleTrustcoin2.sol

```diff
@@ -64,87 +58,68 @@ contract ExampleTrustcoin2 is ERC20TokenInterface, SafeMath {
   }
 
   // See ERC20
-  function balanceOf(address _owner) constant returns (uint balance) {
+  function balanceOf(address _owner) constant external returns (uint balance) {
     return balances[_owner];
   }
 
   // See ERC20
-  function approve(address _spender, uint _value) returns (bool success) {
+  function approve(address _spender, uint _value) external returns (bool success) {
     allowed[msg.sender][_spender] = _value;
     Approval(msg.sender, _spender, _value);
     return true;
   }
 
   // See ERC20
-  function allowance(address _owner, address _spender) constant returns (uint remaining) {
+  function allowance(address _owner, address _spender) constant external returns (uint remaining) {
     return allowed[_owner][_spender];
   }
 
   //
   //  Migration methods
   //
 
-  /**
-   *  Changes the owner for the migration behaviour
-   *  @param _master Address of the migration controller
-   */
-  function changeMigrationMaster(address _master) onlyFromMigrationMaster external {
+  // See MigratedTokenInterface
+  function migrateOldTokens(address _from, uint256 _value) external {
+    if (!allowIncomingMigrations) throw;
+    if (_value == 0) throw;
+    totalSupply = safeAdd(totalSupply, _value);
+    balances[_from] = safeAdd(balances[_from], _value);
+    IncomingMigration(_from, _value);
+  }
+
+  // See MigratableTokenInterface
+  function changeMigrationMaster(address _master) external {
     if (_master == 0) throw;
     migrationMaster = _master;
   }
 
-  /**
-   *  Sets the address of the new token contract, so we know who to
-   *  accept discardTokens() calls from, and enables token migrations
-   *  @param _newTokenAddress Address of the new Trustcoin contract
-   */
-  function setNewTokenAddress(address _newTokenAddress) onlyFromMigrationMaster external {
+  // See MigratableTokenInterface
+  function setNewTokenAddress(address _newTokenAddress) external {
     if (newTokenAddress != 0) throw; // Ensure we haven't already set the new token
     if (_newTokenAddress == 0) throw;
     newTokenAddress = _newTokenAddress;
+    allowOutgoingMigrationsUntil = (now + 26 weeks); // Only allow migrations for the next six months
```

Comment:
26 weeks is 6 months AFAIK, I've made it the lower bound for migration time period too.